.github/actions/javascript/createOrUpdateStagingDeploy/index.js

```diff
@@ -238,26 +238,39 @@ function getMergeLogsAsJSON(fromRef, toRef) {
 /**
  * Parse merged PRs, excluding those from irrelevant branches.
  *
- * @param {Array<String>} commitMessages
+ * @param {Array<Object<{commit: String, subject: String, authorName: String}>>} commits
  * @returns {Array<String>}
  */
-function getValidMergedPRs(commitMessages) {
-    return _.reduce(
-        commitMessages,
-        (mergedPRs, commitMessage) => {
-            if (!_.isString(commitMessage)) {
-                return mergedPRs;
-            }
+function getValidMergedPRs(commits) {
+    const mergedPRs = new Set();
+    _.each(commits, (commit) => {
+        const author = commit.authorName;
+        if (author === 'OSBotify') {
+            return;
+        }
 
-            const match = commitMessage.match(/Merge pull request #(\d+) from (?!Expensify\/(?:main|version-|update-staging-from-main|update-production-from-staging))/);
-            if (!_.isNull(match) && match[1]) {
-                mergedPRs.push(match[1]);
-            }
+        const message = commit.subject;
+        if (!_.isString(message)) {
+            return;
```

Comment:
I find it confusing that `return` here means `continue`. Not sure if its work of a comment line `// continue` but if more people would agree it might be a simple  change to improve the readability of the code. Just depends if you are used to write with the lodash methods.

Reply:
I'm not sure how to respond to that since our code guidelines + eslint config explicitly promote the use of `_.each` over regular `for` loops. I tend to agree things might be simpler and more readable in many cases if we just used regular for loops. Our eslint config disallows `for...of` loops for some vague performance reasons but honestly I would love to re-evaluate that with the TS migration.

In any event, I'm going to say that this convo is out-of-scope for this PR.

Reply:
Brought this up in https://expensify.slack.com/archives/C01GTK53T8Q/p1687477811950459